run/pubsub/main_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestHelloPubSubErrors(t *testing.T) {
+	tests := []struct {
+		name string
+	}{
+		{name: "no_payload"},
+		{name: "not_base64"},
+	}
+	for _, test := range tests {
+		var payload *strings.Reader
```

Comment:
`payload` should be a part of the test struct, right? Also `not_encoded` and `fmt.Sprintf` are not needed below, so it would be fine to inline it all.

Or maybe if you want to keep symmetry with the normal test below, add an `encode` value to the test struct? Relying on the test name feels weird.